bpf/lib/egress_gateway.h

```diff
@@ -29,26 +29,31 @@ int egress_gw_fib_lookup_and_redirect(struct __ctx_buff *ctx, __be32 egress_ip,
 				      __s8 *ext_err)
 {
 	struct bpf_fib_lookup_padded fib_params = {};
-	__u32 old_oif = ctx_get_ifindex(ctx);
+	int oif = 0;
 
 	*ext_err = (__s8)fib_lookup_v4(ctx, &fib_params, egress_ip, daddr, 0);
 
 	switch (*ext_err) {
 	case BPF_FIB_LKUP_RET_SUCCESS:
+		break;
 	case BPF_FIB_LKUP_RET_NO_NEIGH:
-		if (old_oif == fib_params.l.ifindex)
+		/* Don't redirect if we can't update the L2 DMAC: */
+		if (!neigh_resolver_available())
+			return CTX_ACT_OK;
+
+		/* Don't redirect without a valid target ifindex: */
+		if (!is_defined(HAVE_FIB_IFINDEX))
```

Comment:
Note that in practice this doesn't make any actual difference - by bailing out on `neigh_resolver_available()` (a 5.10 kernel feature), we already cover `HAVE_FIB_INDEX` (also introduced in 5.10).